app/screens/home/tab_bar/account.tsx

```diff
@@ -19,22 +20,32 @@ type Props = {
 }
 
 const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
-    selected: {
+    container: {
         borderWidth: 2,
-        borderColor: theme.buttonBg,
         borderRadius: 20,
     },
+    focused: {
+        borderColor: theme.buttonBg,
+    },
+    unfocused: {
+        borderColor: changeOpacity(theme.centerChannelColor, 0.48),
+    },
 }));
 
-const Account = ({currentUser, isFocused, theme}: Props) => {
+export const Account = ({currentUser, isFocused, theme}: Props) => {
```

Comment:
I am not a big fan of exporting stuff just for testing, since they can be inadvertently imported.

My approach is either test everything on the exported interface, or export a new object (`exportsForTest` or something like that) with all the elements you want to export only for testing purposes.

It is dirty and ugly, but helps not importing things that shouldn't be imported in other files.

Reply:
Having two different files seems more "the way we do things". It still opens the door to importing the wrong component, but it is less common (it is easier to notice that you are importing something like: `components/myComponent/myComponent`).

I don't disagree with your claim of "testing in isolation and in the most atomic level", but it is something to be careful about, since it can be driven to the absurd. For example, you could consider that each callback is an atomic part, and should be tested separately, and your component will look like the following:
```
export const myCallback1 = (prop1, prop2, state1, state2, setState1, setState2) => {...};
export const myCallback2 = (prop1, state1, setState2) => {...};

const MyComponent = (prop1, prop2) => {
    const [state1, setState1] = useState();
    const [state2, setState2] = useState();
    const c1 = useCallback(() => myCallback1(prop1, prop2, state1, state2, setState1, setState2));
    const c2 = useCallback(() => myCallback2(prop1, state1, setState2));
    return (...);
}
export default MyComponent;
```

With your approach here, you are creating a new "implementation detail separation" between the component and the with observable. And what I mean with an "implementation detail separation" is that the component will never be used without the with observable. Even if in code they can be separate, logically they are an inseparable atomic thing.

That being said, that approach can also gotten to the absurd and say that the whole app is an atomic inseparable thing, and we should only the test the app as a whole.

In summary... I don't disagree on making this cut as long as it makes testing easier, but as a rule of thumb I would try to make the test of the whole component (state and component, not state on one side and component on the other).

Reply:
Understood.  I see your points.  I have been invited and accepted to join the testing team and will bring this up to see where we can find common grounds.  In the meantime, I will leave it as it is, with the knowledge that Account now can be imported as a component without observables, and eventually it should and it could (as part of design components).